### ISSUE TITLE
FEProblemBase::_dt should be recoverable data?

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#11198]
## Rationale

It is currently a restartable data. The evidence is that if I change it to `declareRecoverableData` all framework tests pass. Also `Transient` executioner reevaluate `_dt` in its `init()` during restart.

## Description

A small issue can be used to start careful examinations on the recoverable and restartable data. At least I did not pay attention on the difference in the past. Looks like it does not hurt to declare the data as restartable but doing recover calculation.

## Impact
Better code.

**Comment by permcody** [comment]
@friedmud had a specific design here. It probably needs to be documented as to what he was after. I'd be curious to see the PR to see what breaks. However, a PR probably isn't good enough. We'd have to check the sprint problem and other more complex examples and real problems.

**Comment by YaqiWang** [comment]
Agree that this kind of changes needs to be extremely careful. I just feel lots of restartable data can be just recoverable, otherwise they are loaded during initial setup but then discarded by construction and setup. For example, https://github.com/idaholab/moose/blob/devel/framework/src/timesteppers/TimeSequenceStepperBase.C#L24-L25. One thing I did not check in the code: is recoverable data always also restartable data? I knew the other way around is not true. I think we can and should create a new tester to make sure the integrity of checkpoint files.

**Comment by friedmud** [comment]
It is restartable on purpose - you can use `reset_dt` in the `Executioner` block to override this behavior.

I have been over and over all of the scenarios.  The defaults for `_time` and `_dt` are this way on purpose.

**Comment by YaqiWang** [comment]
You did not give an objective reason for closing this issue.

**Comment by YaqiWang** [comment]
Oh, sorry, I misunderstood the parameter `reset_dt` what you said about. Can somebody answer me this: is recoverable data always also restartable data? i.e. when we do recovery, restartable data will be loaded?

**Comment by friedmud** [comment]
All restartable data is recoverable.

Restartable is a subset of recoverable.

**Comment by permcody** [comment]
@friedmud - I was working on fixing up start_time for the NRC and wanted to document the restart behavior of dt and using reset_dt and ran across this ticket. The `reset_dt` behavior works a little too well in the sense that you cannot do anything but reset the dt in current MOOSE! Yes there is a test for reset_dt but it behaves exactly the same if you remove that parameter or explicitly set it to false so it's just broken.

The problem is that in the TimeStepper the choose on whether to use initialDT or a computed dt is based on the time step, which is declared as "recoverable" not "restartable". So - it's worth talking about what we should do here. I'm sure I could change the logic to make dt not depend on t_step, or we could drop that option altogether since it doesn't work. I'm open to ideas. With my new PR (#13188) start_time will start working properly in that you can either set it, or not on restart.

**Comment by friedmud** [comment]
If you have a TimeStepper that is using time_step to set the dt... then it
doesn't care about reset_dt.

What reset_dt is there for is for _adaptive_ timestepping algorithms.  For
instance, IterationAdaptive or DT2 or whatever.  In those cases you might
want to force your new simulation to "restart" the adaptive timestepping.

That said: reset_dt is definitely a kludge (I needed it for something).  So
I'm down for a real fix...

**Comment by permcody** [comment]
Yeah - I figured that was your design. The problem is that the logic to
reset DT is hardcoded into the TimeStepper base class and while this method
is overridable, I don't think there's a single Timestepper that overrides
that method.

That's fine, just wanted your thought. It's possible that the design of the
Timesteppers class has evolved since it was originally created and this
feature just got lost along the way. It's not urgent to fix it, just noting
that it doesn't currently work.

**Comment by lindsayad** [comment]
I'm changing the title of this ticket since what we regard as "the issue" has evolved with its closing and reopening
